redhat-appstudio/service-provider-integration-operator
# Initial batch of Vault-related metrics

### What does this PR do?
This creates a number of metrics for measuring the access times to Vault.

### Screenshot/screencast of this PR
```
$ kubectl -n spi-system exec deployment/spi-controller-manager -c manager -t -- curl --silent localhost:8080/metrics
...
# HELP redhat_appstudio_spi_vault_request_count_total The request counts to Vault categorized by status code
# TYPE redhat_appstudio_spi_vault_request_count_total counter
redhat_appstudio_spi_vault_request_count_total{status="200"} 24
redhat_appstudio_spi_vault_request_count_total{status="404"} 4
# HELP redhat_appstudio_spi_vault_response_time_seconds The response time of Vault requests categorized by HTTP method and status code
# TYPE redhat_appstudio_spi_vault_response_time_seconds histogram
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="0.005"} 8
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="0.01"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="0.025"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="0.05"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="0.1"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="0.25"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="0.5"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="1"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="2.5"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="5"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="10"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="200",le="+Inf"} 24
redhat_appstudio_spi_vault_response_time_seconds_sum{method="GET",status="200"} 0.127462067
redhat_appstudio_spi_vault_response_time_seconds_count{method="GET",status="200"} 24
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="0.005"} 3
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="0.01"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="0.025"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="0.05"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="0.1"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="0.25"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="0.5"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="1"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="2.5"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="5"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="10"} 4
redhat_appstudio_spi_vault_response_time_seconds_bucket{method="GET",status="404",le="+Inf"} 4
redhat_appstudio_spi_vault_response_time_seconds_sum{method="GET",status="404"} 0.016550357999999998
redhat_appstudio_spi_vault_response_time_seconds_count{method="GET",status="404"} 4
...
```

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-218

### How to test this PR?
1. Use the `quay.io/lkrejci/service-provider-integration-operator:metrics` image for running the operator
2.  Check that there is many metrics returned:
``` 
   kubectl -n spi-system exec deployment/spi-controller-manager -c manager -t -- curl --silent localhost:8080/metrics | \
   grep  redhat_appstudio_spi | \
   wc -l
```
